### PR TITLE
factor out common call graph unit test functions into header

### PR DESCRIPTION
**Pull request by polgreen**

### CODE REVIEW

File: unit/analyses/call_graph.cpp [hunk 1]
```diff
@@ -14,34 +14,10 @@ Module: Unit test for call graph generation
...
-  const irep_idt &key,
-  const std::set<irep_idt> &values)
-{
-  auto matching_values=map.equal_range(key);
-  std::set<irep_idt> matching_set;
-  for(auto it=matching_values.first; it!=matching_values.second; ++it)
-    matching_set.insert(it->second);
-  return matching_set==values;
-}
+#include "call_graph_test_utils.h"
```
Comment (tautschnig) [inline unit/analyses/call_graph.cpp]: There is `unit/testing-utils/` and I'm wondering whether that would be a better place to put it? Along with that I'd ask for a cpp file to be added that will take the definitions while the header file would just have the declarations.

**Review by tautschnig — APPROVED** [review]
Approving under the assumption that the below change will be implemented. Possibly somewhat relaxed standards apply for test set-ups, but I think the below ask is not too unreasonable.

**Review by smowton — APPROVED** [review]
LGTM once dependent PRs merged and rebased

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 19a1ceb0bac1974cfc83f0991abf5592f10a466f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77464017

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 2431ac05071cd71963f29a412885e90e7676634c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77466912

**Comment by polgreen** [comment]
now rebased

**Pull request merged**